### PR TITLE
Plugin loader throws for plugins with glaring issues

### PR DESCRIPTION
**Pull request by kjin**
Throw for plugin patches
- with both `patch` and `intercept`
- with neither

Warn for plugin patches
- with `unpatch` and `intercept` (should this throw as well?)

Open to any thoughts regarding severity level.

### CODE REVIEW

File: src/trace-plugin-loader.js [hunk 1]
```diff
@@ -49,6 +49,19 @@ function checkLoadedModules() {
   }
 }
 
+function checkPatch(patch) {
+  if (!patch.patch && !patch.intercept) {
+    throw new Error('Plugin patch for ' + patch + ' doesn\'t patch ' +
```
Comment (matthewloring) [inline src/trace-plugin-loader.js]: This message could be ambiguous. Maybe "Plugin for ' + patch + ' doesn\'t contain any patches."?

File: src/trace-plugin-loader.js [hunk 2]
```diff
@@ -49,6 +49,19 @@ function checkLoadedModules() {
...
+      'anything.');
+  } else if (patch.patch && patch.intercept) {
+    throw new Error('Plugin patch for ' + patch.file + ' has ' +
+      'both intercept and patch functions.');
+  } else if (patch.unpatch && patch.intercept) {
```
Comment (matthewloring) [inline src/trace-plugin-loader.js]: I think having a warnings for patch without unpatch and unpatch without patch might be good. I think with these combinations a separate warning for unpatch with intercept is obsolete.

Comment (kjin) [inline src/trace-plugin-loader.js]: Added for `patch.patch && !patch.unpatch` (the latter case is already covered).

File: src/trace-plugin-loader.js (hunk 1 above)
Comment (kjin) [inline src/trace-plugin-loader.js]: Done (also corrected `patch` to `patch.file`)

**Pull request merged**